pkg/transaction/monitor.go

```diff
@@ -169,44 +171,34 @@ func (tm *transactionMonitor) watchPending() {
 	}
 }
 
-// potentiallyConfirmedTxWatches returns all watches with nonce less than what was specified
-func (tm *transactionMonitor) potentiallyConfirmedTxWatches(nonce uint64) (watches map[uint64]map[common.Hash][]transactionWatch) {
+func (tm *transactionMonitor) hasWatches() bool {
 	tm.lock.Lock()
 	defer tm.lock.Unlock()
+	return len(tm.watchesByNonce) > 0
+}
 
-	potentiallyConfirmedTxWatches := make(map[uint64]map[common.Hash][]transactionWatch)
-	for n, watches := range tm.watchesByNonce {
-		if n < nonce {
-			potentiallyConfirmedTxWatches[n] = watches
+func watchStart(watches []transactionWatch) time.Time {
+	var start time.Time
+	for _, w := range watches {
+		if w.start.Before(start) {
+			start = w.start
```

Comment:
is it possible to have a `w.start` that is 0? because when we append these watches, the start is initialized with `time.Now()`

Reply:
In the scope of a function, zero value is possible. In the scope of the package, currently it is not.